### DISCUSSION TITLE
Reducing duplication for steps that run for PRs and master

### DISCUSSION THREAD

**Discussion by leonardehrenfried** [community/community#25329] [Actions]
<p>I have a pretty common workflow.</p>
<p>I want to run some tests for each pull request and after the merge to master I run the tests again and then deploy an artifact, in my instance a docker container.</p>
<p>So far I have managed to only achieve this by creating two workflow files, <a href="https://github.com/stadtnavi/digitransit-ui/blob/0e6f70132e0fd5357d4cd088f435c2ccd4d57e58/.github/workflows/docker.yml" rel="nofollow noopener">one for the master branch</a>, <a href="https://github.com/stadtnavi/digitransit-ui/blob/0e6f70132e0fd5357d4cd088f435c2ccd4d57e58/.github/workflows/tests.yml" rel="nofollow noopener">one for the PRs</a>.</p>
<p>If you check the files, you can see that the test step is completely identical.</p>
<p>How can I achieve the same thing without having multiple copies of the test step?</p>
<p>Can I add references to a common step from both files? Is there another way I haven’t considered?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/leonardehrenfried">@leonardehrenfried</a>,</p>
<p>Glad to see you in Github Community Forum!<br>
Add references to a common step from both files is not supported. The two workflows are different, based on your yaml code, it will:</p>
<ol>
<li>Trigger pull request workflow and test when create a PR to master branch.</li>
<li>
<strong>Besides the merge from PR, any commits in master branch</strong> will trigger the tests and then deploy an artifact.</li>
</ol>
<p>If this is your expectation, you have to copy the test step into two yaml files, since test could be failed in the pull request workflow which will block the merge, and you need to execute test for the direct commit in master branch as well.</p>
<p>Thanks.</p>

**Comment by airtower-luna** [comment]
<p>Would it be an option for you to remove the branch restrictions on the workflows, and instead check on the <code>docker</code> job (using <code>if:</code> either on the job or relevant steps) if you’re running on <code>master</code>, and skip if not?</p>

**Comment by leonardehrenfried** [comment]
<p>This was exactly the sort of trick that i’m looking for. Can you give an example of what the syntax looks like?</p>

**Comment by leonardehrenfried** [comment]
<p>Is it perhaps:</p>
<pre><code class="lang-auto">if: ${{ github.ref == "master" }}
</code></pre>
<p>?</p>

**Comment by airtower-luna** [comment]
<p>I think so, but I haven’t tried it myself. But as I understand the documentation adding that to the job should work. I’d recommend trying with a test job on a test branch (and a second test branch that shouldn’t run that job).</p>

**Comment by leonardehrenfried** [comment, marked as answer]
<p>I figured it out. The correct syntax is:</p>
<pre><code class="lang-auto">if: ${{ github.ref == 'refs/heads/master' }}
</code></pre>
<p>Here is a complete example: <a href="https://github.com/stadtnavi/hsl-map-server/blob/6aadfc70df0f60ae4f68980ca546af788c6fb37c/.github/workflows/docker.yml" rel="nofollow noopener">https://github.com/stadtnavi/hsl-map-server/blob/6aadfc70df0f60ae4f68980ca546af788c6fb37c/.github/workflows/docker.yml</a></p>

**Comment by airtower-luna** [comment]
<p>Nice, thank you for sharing the working code!</p>
